python/docs/source/user_guide/sql/python_data_source.rst

```diff
@@ -59,8 +59,17 @@ Start by creating a new subclass of :class:`DataSource`. Define the source name,
         def reader(self, schema: StructType):
             return FakeDataSourceReader(schema, self.options)
 
+        def streamReader(self, schema: StructType):
```

Comment:
Given they will see the example from the start, it could give a confusion that every data source has to implement both batch and streaming. Probably better to have some explanation.